libs/designsystem/src/lib/components/page/page.component.scss

```diff
@@ -23,6 +27,41 @@ ion-toolbar {
   --padding-top: 0;
   --padding-bottom: 0;
 
+  // The toolbar height for iPhone's with a notched display.
+  $toolbar-height-notch: utils.size('xxl');
+
+  // The toolbar height for other displays.
+  $toolbar-height-default: utils.size('xxxl');
+
+  // An iPhone with a ios-safe-area-top exceeding 32px is considered to have notched display.
+  $notch-safe-area-top-trigger: utils.size('l');
+
+  // Notch indicator:
+  // 0px represents FALSE
+  // 1px represents TRUE
+  // IF `--kirby-safe-area-top` is defined _and_ has a value strictly larger than `$notch-safe-area-top-trigger`
+  // THEN `--notch-present` = 1px (a notch _is_ present)
+  // ELSE `--notch-present` = 0px (no notch)
+  --notch-present: clamp(
+    0px,
+    var(--kirby-safe-area-top, 0px) - #{$notch-safe-area-top-trigger},
+    1px
+  );
+
+  // Equivalent to:
+  // IF --notch-present == 1px (i.e. the display _does_ have a notch)
+  // THEN --min-height = $toolbar-height-notch
+  // ELSE --min-height = $toolbar-height-default
+  --min-height: clamp(
+    #{$toolbar-height-notch},
+    (1px - var(--notch-present)) * #{utils.strip-unit($toolbar-height-default)},
+    #{$toolbar-height-default}
+  );
```

Comment:
There is definitely significant overhead in parsing this, I used all available brainpower (might not be much, but still!) for a couple of minutes before this was clear to me. 

But the comments helps **a lot**, and while it feels like one of those good ideas that might just be a hack, I think it is the best option.

Good work!

Reply:
Great - point taken 👍